backend/src/main/java/com/woowacourse/momo/service/attendee/AttendeeService.java

```diff
@@ -0,0 +1,50 @@
+package com.woowacourse.momo.service.attendee;
+
+import com.woowacourse.momo.auth.JwtManager;
+import com.woowacourse.momo.domain.attendee.Attendee;
+import com.woowacourse.momo.domain.attendee.AttendeeName;
+import com.woowacourse.momo.domain.attendee.AttendeePassword;
+import com.woowacourse.momo.domain.attendee.AttendeeRepository;
+import com.woowacourse.momo.domain.attendee.Role;
+import com.woowacourse.momo.domain.meeting.Meeting;
+import com.woowacourse.momo.domain.meeting.MeetingRepository;
+import com.woowacourse.momo.exception.MomoException;
+import com.woowacourse.momo.exception.code.AttendeeErrorCode;
+import com.woowacourse.momo.service.attendee.dto.AttendeeLoginRequest;
+import com.woowacourse.momo.service.attendee.dto.TokenResponse;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class AttendeeService {
+
+    private final AttendeeRepository attendeeRepository;
+    private final MeetingRepository meetingRepository;
+    private final JwtManager jwtManager;
+
+    @Transactional
+    public TokenResponse login(String uuid, AttendeeLoginRequest request) {
+        Meeting meeting = meetingRepository.findByUuid(uuid)
+                .orElseThrow(() -> new MomoException(AttendeeErrorCode.INVALID_UUID));
+
+        AttendeeName name = new AttendeeName(request.name());
+        AttendeePassword password = new AttendeePassword(request.password());
+
+        return attendeeRepository.findByMeetingAndName(meeting, name)
+                .map(attendee -> handleExistingAttendee(attendee, password))
+                .orElseGet(() -> handleNewAttendee(meeting, name, password));
+    }
+
+    private TokenResponse handleExistingAttendee(Attendee attendee, AttendeePassword password) {
+        attendee.verifyPassword(password);
+        return new TokenResponse(jwtManager.generate(attendee));
+    }
+
+    private TokenResponse handleNewAttendee(Meeting meeting, AttendeeName name, AttendeePassword password) {
+        Attendee attendee = new Attendee(meeting, name, password, Role.GUEST);
+        attendeeRepository.save(attendee);
+        return new TokenResponse(jwtManager.generate(attendee));
+    }
```

Comment:
[질문] 두 코드를 비교해서 뭐가 나은지 답변 부탁드립니다!
```suggestion
        Optional<Attendee> optionalAttendee = attendeeRepository.findByMeetingAndName(meeting, name);
        if (optionalAttendee.isPresent()) {
            return handleExistingAttendee(optionalAttendee, password);
        }
        return handleNewAttendee(meeting, name, password);
    }

    private TokenResponse handleExistingAttendee(Optional<Attendee> optionalAttendee, AttendeePassword password) {
        Attendee attendee = optionalAttendee.get();
        attendee.verifyPassword(password);
        return new TokenResponse(jwtManager.generate(attendee));
    }

    private TokenResponse handleNewAttendee(Meeting meeting, AttendeeName name, AttendeePassword password) {
        Attendee attendee = new Attendee(meeting, name, password, Role.GUEST);
        attendeeRepository.save(attendee);
        return new TokenResponse(jwtManager.generate(attendee));
    }
```

Reply:
비회원 로그인 시 
`존재하지 않는 참가자 이름`이면 회원가입 및 토큰 발행
`존재하는 참가자 이름`이면 비밀번호를 확인하고 토큰을 발행합니다.

비즈니스 요구사항에 따라 두 가지 관심사를 한 메서드에서 처리하다보니
필연적으로 분기가 발생합니다. 둘 중에 뭐가 나은지 고민되네요.

Reply:
`Optional.isPresent()`를 if 조건에서 검사하여 분기하는 방식은 권장되지 않습니다.  
현재 코드가 더 나아 보여요. 단, `handle~~()` 메서드 이름은 좀 더 명확하게 변경되는 게 좋을 것 같네요!

Reply:
안티 패턴이긴한데 분기를 처리하는 방법으로 저게 최선이었던 것 같네요 ㅋㅋ
현재 코드는 메서드 체이닝이 좀 과한감이 있나 싶었는데, 대안이 없다면 이대로 두겠습니다.

Reply:
안티 패턴이긴한데 분기를 처리하는 방법으로 저게 최선이었던 것 같네요 ㅋㅋ
현재 코드는 체이닝이 좀 과하지 않나 싶었는데, 딱히 대안이 없어 이대로 두겠습니다.

Reply:
메서드 이름은 계속 생각해볼게요 😎

Reply:
저도 페드로의 의견에 동의합니다! 이 상태로 유지하되, 메서드 이름을 좀 더 명확하게 변경하면 좋을 것 같아요~

현재 로직은 
* `attendee`가 존재하면 `attendee`의 비밀번호를 검증한 뒤 `TokenResponse`를 생성하여 반환
*  `attende`가 존재하지 않는다면 `attendee`를 회원으로 등록한 뒤 `TokenResponse`를 생성하여 반환
이렇게 나눠지니까, `handleExistingAttendee` 메서드 명을 `verifyPassword`로, `handleNewAttendee`를 `signUp` 으로 두면 어떨까요? 지금 당장은 이렇게 생각나네요..ㅎ